design/one-pager-resource-connectivity-mvp.md

```diff
@@ -638,7 +638,279 @@ parameters:
+2. **Multiple managed resources**: Create a corresponding managed resource for each required external resource, and manage their connectivity by cross-object references.
+
+Although the first approach requires potentially less effort, the latter approach provides a few advantages:
+
+1. **Reusability**: the new managed resource types could be later re-used for other configurations
+1. **YAML Configuration**: instead of implementing a specific configuration logic inside `AwsNetwork`, the configuration details are expressed in a YAML file. This makes creating more sophisticated configurations possible, without having to write controllers. 
+
+This document focuses on implementing the second approach. For doing so, we address the following challenges:
+
+* As pointed out [earlier](#resource-references) in this document, each managed resource can refer to other resources either by their external names, or their object reference. In AWS case, since we are taking a YAML configuration approach, interconnecting resource by api objects makes a lot of sense. At the same time it is still useful to re-use existing external resources. Because of these requirements, we introduce the notion of *reference type* as following:
+
+	```yaml
+	...
+	spec:
+	  ...
+	  someParameter:
+	    # one of {external, apiObject} values
+	    referenceType: apiObject
+      # a string of attributeName.namespacedName format (apiObject case), or arbitrary (external case)
+	    value: attributeNAme.sampleObject.sampleNamespace
+	  ...
+	...
+	```
+	This indicates that the value for `someParameter` will be retrieved from the given attribute, of the given api object. If the `referenceType` was `external`, then the `value` would be interpreted as the actual value of the external resources's requested attribute.
+* One of the issues that rises in this approach, is the dependency that a managed resource might take on another managed resource, when the former uses the latter in its spec. For example, in order to create a `AwsSubnet` instance which consumes a `AwsVpc` instance, the `AwsVpc` instance has to be created and ready. To solve this, we introduce the notion of *resource dependency* in consuming resource's spec:
+	```yaml
+	...
+	metadata:
+	  - namespace: example
+	    name: sample
+	...
+	spec:
+	  ...
+    dependsOn:
+      # a list of strings, each in namespacedName format
+  		- requiredResourceName.requiredResourceNamespace
+	  ...
+	...
+	```
+
+	This indicates that `sample.example` resource needs to wait until all required resources are ready. Moreover, if a resource refers to another resource as an `apiObject` explained above, its dependency is implicit and there is no need to provide the `dependsOn` explicitly in the spec. This is the case in the following resource configurations.
```

Comment:
What happens if we omit this functionality?

I imagine that if managed resource `foo` references and depends upon external resource `bar`, and `bar` does not exist at `foo` creation time, `foo` will simply fail to reconcile, updating its `ConditionedStatus` to reflect that it cannot be created because `bar` does not exist. When `bar` is created `foo` will successfully reconcile.

Reply:
Please see my response to dan's similar comment:
https://github.com/crossplaneio/crossplane/pull/700#discussion_r316931978